config: minor credential upgrades (#8997)
tarantool/tarantool: changelogs/unreleased/gh-8967-upgrade-password-sync.md

## feature/config

> * Implemented full password support in config.credentials schema,
>   including password set, update, removal for `chap-sha1` auth type
>   (supported by both Tarantool Community Edition and Tarantool Enterprise
>   Edition) and `pap-sha256` (just for Enterprise Edition where it is
>   available) (gh-8967).

Reviewer: ```suggestion
* Implemented a full password support in the `config.credentials` schema,
  including a password setting, updating, and removal for the `chap-sha1` auth type
  (supported by both Tarantool Community Edition and Tarantool Enterprise
  Edition) and `pap-sha256` (just for Enterprise Edition where it is
  available) (gh-8967).
```

Author: Applied.
